### PR TITLE
The world is not ready for esm yet

### LINKED ISSUE #4: Add API to manually change color mode
PostCSS, Logux Server and some other Nano Colors users have own options to change colors.

Can we have a way to manually change support mode like:

```js
import { createColor } from 'picocolors'

let { red } = createColor(options.color)
```

Or:

```js
import { setColors, isColorSupported } from 'picocolors'

setColors(options.color ?? isColorSupported)
```

### LINKED ISSUE #13: Add `default` export to CJS
`picocolors.browser.cjs` and other `cjs` should have `default` export to be ready for bugs like https://github.com/postcss/autoprefixer/issues/1425

For instance, `module.exports.default = module.exports` will fix the case.

### PR DESCRIPTION
**Pull request by alexeyraspopov**
## Rationale

With this PR we break any possible compatibility with nanocolors and colorette for the sake of larger ecosystem support, smaller package size, and the ability to enforce color (via custom API).

- Supporting ESM-first is just too much pain (#25, #2). Maybe the story will change in a year or two
- Tree-shaking is not beneficial for a library which primary use is Node.js CLI tooling
- Import of an API object is more flexible than adding every single formatting function to the import

Additionaly, I decided to make another change here, while I'm at it: _using tabs instead of spaces_. Because I like tabs with size 2 and it also reduces package size. The PR includes `.editorconfig` so everyone who's using the plugin should be fine. Also, GitHub properly supports it on the website.

Preliminary checking the package size (via `clean-publish` and `npm publish --dry-run`) shows following stats:

```
package size:  2.4 kB
unpacked size: 5.6 kB
```

That's a win-win

Closes #4
Closes #13
Closes #22
Closes #25

### CODE REVIEW

File: picocolors.d.ts [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+	gray: (input: string) => string,
+	bgBlack: (input: string) => string,
+	bgRed: (input: string) => string,
+	bgGreen: (input: string) => string,
+	bgYellow: (input: string) => string,
+	bgBlue: (input: string) => string,
+	bgMagenta: (input: string) => string,
+	bgCyan: (input: string) => string,
+	bgWhite: (input: string) => string,
+};
```
Comment (ai) [inline picocolors.d.ts:28]: Need `createColors` type too.

**Comment by alexeyraspopov** [comment]
@ai, I made another attempt to do `.d.ts` but I need someone to verify it since I'm not quite good at it

**Comment by ai** [comment]
@alexeyraspopov `.d.ts` looks good, but `createColors`’s result doesn’t have `createColors`. It is better to use:

```ts
export type Formatter = (input: string | number | null | undefined) => string;

export interface Colors {
	isColorSupported: boolean;
	reset: Formatter;
	bold: Formatter;
	dim: Formatter;
	italic: Formatter;
	underline: Formatter;
	inverse: Formatter;
	hidden: Formatter;
	strikethrough: Formatter;
	black: Formatter;
	red: Formatter;
	green: Formatter;
	yellow: Formatter;
	blue: Formatter;
	magenta: Formatter;
	cyan: Formatter;
	white: Formatter;
	gray: Formatter;
	bgBlack: Formatter;
	bgRed: Formatter;
	bgGreen: Formatter;
	bgYellow: Formatter;
	bgBlue: Formatter;
	bgMagenta: Formatter;
	bgCyan: Formatter;
	bgWhite: Formatter;
}

export = Colors & { createColors: (enabled: boolean) => Colors };
```

Also, it is useful for some TS projects to be able to re--use `Formatter` and `Colors` types.

File: README.md [hunk 2]
```diff
@@ -5,20 +5,19 @@
 A tinier and faster alternative to [nanocolors](https://github.com/ai/nanocolors). Andrey, are you even trying?
 
 ```javascript
-import { green, italic } from "picocolors";
+import pc from "picocolors";
```
Comment (TrySound) [inline README.md:8]: Explicit commonjs exports allows to use named imports in node
```js
exports.red = red;
// usage
import { red } from 'picocolors';
```

**Pull request merged**